geode-core/src/main/java/org/apache/geode/internal/cache/tx/RemoteOperationMessage.java

```diff
@@ -177,7 +179,28 @@ public void process(final ClusterDistributionManager dm) {
       sendReply(getSender(), this.processorId, dm, replyException, null, 0);
       return;
     }
-    dm.getExecutors().getWaitingThreadPool().execute(() -> doRemoteOperation(dm, cache));
+
+    String conserveSockets = getConserveSocketsSetting(dm);
+    if (conserveSockets != null && conserveSockets.equals("false")) {
+      // reply inline for CONSERVE_SOCKETS == false case.
+      doRemoteOperation(dm, cache);
+      return;
+    }
+
+    if (isTransactional()) {
+      dm.getExecutors().getWaitingThreadPool().execute(() -> doRemoteOperation(dm, cache));
+    } else {
+      // reply inline for non-transactional case.
+      doRemoteOperation(dm, cache);
+    }
+  }
+
+  String getConserveSocketsSetting(ClusterDistributionManager dm) {
+    return dm.getSystem().getProperties().getProperty(CONSERVE_SOCKETS);
```

Comment:
I think it would be better to get the value of `conserve-sockets` from `DistributionConfigImpl` rather than check if a system property has been set, since a user may be using the default value. Since the default value of `conserve-sockets` was changed in 1.14, this would lead to a difference in behaviour when checking if the system property is present or not. If that's not possible, then a check should be included to see what the default value of `conserve-sockets` is (`DistributionConfig.DEFAULT_CONSERVE_SOCKETS`) if the system property is not set so that the correct action can be taken.

Reply:
I agree with Donal, though this isn't a System Property but a DistributionConfig converted into properties.  It would be better to get the config and query it.  Creating a Properties object isn't all that cheap.

Reply:
I think what you want to know here is whether the message is going to be sent on a thread-owned Socket or a shared Socket.  To determine that you need to use InternalDistributedSystem.threadOwnsResources(), not the conserve-sockets setting.